x-pack/plugins/security_solution/server/lib/machine_learning/index.ts

```diff
@@ -49,6 +52,17 @@ export const getAnomalies = async (
                 },
               },
             ],
+            must_not: buildExceptionFilter({
```

Comment:
Yeah, the logic around those conditions is to prevent an empty object from showing up in the queries, like
```
{
  bool: {}
}
```
Here if the exception filter comes back `undefined` then `must_not` will be omitted from the final query that gets sent to ES, but the `filter` inside `bool` will always exist so we won't end up with any empty objects.